src/Provider/User/SymfonyUserProvider.php

```diff
@@ -10,21 +12,33 @@ final class SymfonyUserProvider implements UserProviderInterface
     /**
      * User for anonymous.
      */
-    const USER_ANONYMOUS = 'anonymous';
+    public const USER_ANONYMOUS = 'anonymous';
 
     /**
      * Value for user when we are in cli.
      */
-    const USER_CLI = 'cli';
+    public const USER_CLI = 'cli';
 
     /**
      * @var TokenStorageInterface|null
      */
     private $tokenStorage;
 
-    public function __construct(TokenStorageInterface $tokenStorage = null)
+    /**
+     * @var PropertyAccessorInterface
+     */
+    private $propertyAccessor;
+
+    /**
+     * @var string
+     */
+    private $propertyName;
+
+    public function __construct(TokenStorageInterface $tokenStorage = null, PropertyAccessorInterface $propertyAccessor = null)
     {
         $this->tokenStorage = $tokenStorage;
+        $this->propertyAccessor = $propertyAccessor ?: PropertyAccess::createPropertyAccessor();
+        $this->propertyName = 'username';
     }
 
     public function getUser(): string
```

Comment:
What do you think about replacing the setter by an optional argument `propertyName` on `getUser`:

```php
public function getUser(string $propertyName = 'username'): string;
```